protocols/request-response/src/handler.rs

```diff
@@ -68,21 +69,25 @@ where
     /// Queue of events to emit in `poll()`.
     pending_events: VecDeque<Event<TCodec>>,
     /// Outbound upgrades waiting to be emitted as an `OutboundSubstreamRequest`.
-    outbound: VecDeque<RequestProtocol<TCodec>>,
-    /// Inbound upgrades waiting for the incoming request.
-    inbound: FuturesUnordered<
-        BoxFuture<
-            'static,
-            Result<
-                (
-                    (RequestId, TCodec::Request),
-                    oneshot::Sender<TCodec::Response>,
-                ),
-                oneshot::Canceled,
-            >,
-        >,
-    >,
+    pending_outbound: VecDeque<OutboundMessage<TCodec>>,
+
+    requested_outbound: VecDeque<OutboundMessage<TCodec>>,
```

Comment:
I think if you are not using 0RTT stream negotiation, you'll pay a lot of round-trips if you do these streams sequentially because `Connection` won't be giving you a stream until it is fully negotiated.

I think I prefer the implementation this way because it doesn't assume anything about the muxers. The user may request a series of streams in a row so it makes sense to just forward that instead of handling it one by one.

Reply:
Makes sense. Thank you @thomaseizinger for expanding on this.